### PR TITLE
Allowing agencies of any system to have a Capacity and Costs dashboard (as long as they are allowlisted)

### PR DESCRIPTION
**Pull request by lilidworkin**
## Description of the change

Allow agencies of any system to have a dashboard (with capacity and costs metrics only)

This will unblock CSG since they need a Law Enforcement demo tomorrow. They can use the Law Enforcement [DEMO] agency on staging, which @morden35 just added fake data to!

Playtesting: https://lilitest---agency-dashboard-web-b47yvyxs3q-uc.a.run.app/agency/law-enforcement-%5Bdemo%5D

## Type of change

> All pull requests must have at least one of the following labels applied (otherwise the PR will fail):

| Label                       	| Description                                                                                               	|
|-----------------------------	|-----------------------------------------------------------------------------------------------------------	|
| Type: Bug                   	| non-breaking change that fixes an issue                                                                   	|
| Type: Feature               	| non-breaking change that adds functionality                                                               	|
| Type: Breaking Change       	| fix or feature that would cause existing functionality to not work as expected                            	|
| Type: Non-breaking refactor 	| change addresses some tech debt item or prepares for a later change, but does not change functionality    	|
| Type: Configuration Change  	| adjusts configuration to achieve some end related to functionality, development, performance, or security 	|
| Type: Dependency Upgrade      | upgrades a project dependency - these changes are not included in release notes                             |

## Related issues

Closes https://github.com/Recidiviz/justice-counts/issues/851

### Development

**This box MUST be checked by the submitter prior to merging**:
- [X] **Double- and triple-checked that there is no Personally Identifiable Information (PII) being mistakenly added in this pull request**

These boxes should be checked by the submitter prior to merging:
- [ ] Tests have been written to cover the code changed/added as part of this pull request

### Code review

These boxes should be checked by reviewers prior to merging:

- [ ] This pull request has a descriptive title and information useful to a reviewer
- [ ] This pull request has been moved out of a Draft state, has no "Work In Progress" label, and has assigned reviewers
- [ ] Potential security implications or infrastructural changes have been considered, if relevant

### CODE REVIEW

File: agency-dashboard/src/AgencyOverview/AgencyOverview.tsx [hunk 1]
```diff
@@ -86,6 +80,7 @@ export const AgencyOverview = observer(() => {
   useAsyncEffect(async () => {
     try {
       await agencyDataStore.fetchAgencyData(slug as string);
+      setCurrentSystem(agencyDataStore.agency?.systems[0]);
```
Comment (lilidworkin) [inline agency-dashboard/src/AgencyOverview/AgencyOverview.tsx:83]: @mxosman does this seem like a reasonable way to handle the current system state? Could also put it on the store, wasn't sure which was best

**Review by alfonsogoberjr — APPROVED** [review]
This is almost exactly what's in my PR, LGTM

File: agency-dashboard/src/AgencyOverview/AgencyOverview.tsx (hunk 1 above)
Comment (mxosman) [reply agency-dashboard/src/AgencyOverview/AgencyOverview.tsx:83]: Yes - this should work perfectly fine!

**Comment by mxosman** [comment]
This LGTM - not necessarily blocking because it's working on the Law Enforcement and Law Enforcement [DEMO] agency, but the only thing I'm still not 100% sure about is the allow list - I seem to still be able to access some of other agencies (e.g. Department of Mahmoud) - and some times it doesn't show the Capacity and Costs dashboard.

**Comment by lilidworkin** [comment]
Agreed -- something funky going on there. I'm looking into it now too!

**Pull request merged**